include/boost/algorithm/indirect_sort.hpp

```diff
@@ -0,0 +1,192 @@
+	return ret;
+}
+
+/// \fn indirect_stable_sort (RAIterator first, RAIterator last)
+/// \returns a permutation of the elements in the range [first, last)
+///     such that when the permutation is applied to the sequence,
+///     the result is sorted in non-descending order.
+///
+/// \param first  The start of the input sequence
+/// \param last   The end of the input sequence
+///
+template <typename RAIterator>
+Permutation indirect_stable_sort (RAIterator first, RAIterator last) {
+    return indirect_stable_sort(first, last, 
+                    std::less<typename std::iterator_traits<RAIterator>::value_type>());
+}
+
+                    // ===== partial_sort =====
+
+/// \fn indirect_partial_sort (RAIterator first, RAIterator last, Predicate p)
+/// \returns a permutation of the elements in the range [first, last)
+///     such that when the permutation is applied to the sequence,
+///     the resulting range [first, middle) is sorted and the range [middle,last)
+///     consists of elements that are "larger" than then ones in [first, middle),
+///     according to the predicate pred.
+///
+/// \param first  The start of the input sequence
+/// \param middle The end of the range to be sorted
+/// \param last   The end of the input sequence
+/// \param pred   The predicate to compare elements with
+///
+template <typename RAIterator, typename Pred>
+Permutation indirect_partial_sort (RAIterator first, RAIterator middle, 
+                                   RAIterator last, Pred pred) {
+	Permutation ret(std::distance(first, last));
+	
+	boost::algorithm::iota(ret.begin(), ret.end(), size_t(0));
+	std::partial_sort(ret.begin(), ret.begin() + std::distance(first, middle), ret.end(), 
+	                  detail::indirect_predicate<Pred, RAIterator>(pred, first));
+	return ret;
+}
+
+/// \fn indirect_partial_sort (RAIterator first, RAIterator last)
+/// \returns a permutation of the elements in the range [first, last)
+///     such that when the permutation is applied to the sequence,
+///     the resulting range [first, middle) is sorted in non-descending order,
+///     and the range [middle,last) consists of elements that are larger than
+///     then ones in [first, middle).
+///
+/// \param first  The start of the input sequence
+/// \param last   The end of the input sequence
+///
+template <typename RAIterator>
+Permutation indirect_partial_sort (RAIterator first, RAIterator middle, RAIterator last) {
+    return indirect_partial_sort(first, middle, last, 
+                    std::less<typename std::iterator_traits<RAIterator>::value_type>());
+}
+
+                    // ===== nth_element =====
+
+/// \fn indirect_nth_element (RAIterator first, RAIterator last, Predicate p)
+/// \returns a permutation of the elements in the range [first, last)
+///     such that when the permutation is applied to the sequence,
+///     the result is sorted according to the predicate pred.
```

Comment:
Similar C&P mistake in signature and description.

Reply:
Nice catch - thanks!